### ISSUE TITLE
Global translate-cloak

### ISSUE THREAD

**Issue by oktay89** [angular-translate/angular-translate#1241]
I use $translatePartialLoader. I am trying to hide all my untranslated keys till they get translated. This is my code:
CSS
```
.translate-cloak {
    display: none;
}
```
config in app.js
```
    $translateProvider.useLoader('$translatePartialLoader', {
        urlTemplate: '{part}/translations/{lang}.json'
    });
    $translateProvider.preferredLanguage("de");
```
span translate-cloak="key" -> adds a translate-cloak class at load to the element, but never removes it
span translate="key" -> shows the key while loading and then translates it correctly
1. Why translate-cloak class doesn't get removed in the first case?
2. Can translate-cloak be configured globally in app.js?

**Comment by tspaeth** [comment]
Well, if you attach translate-cloak to the body tag for example you'll see that there is a short flickering on visible, not-yet-translated keys. This is probably due to the way how all angular code will be initialized but the dom will already be shown initially.

You can get around this with a small tweak to the directive plus the body tag in my example also:

- attach style="visible: none" to the body tag.
- Add something like
```
tElement[0].style.display = null;
```

right before this:
https://github.com/angular-translate/angular-translate/blob/master/src/directive/translate-cloak.js#L35
and then flickering should be gone but the text will only be displayed after it has been translated.

Hope that helps a bit...

**Comment by knalli** [comment]
@oktay89  Do you use the latest patch release? We had a bug regarding the cloak introduced in the latest minor release.

Additionally: You can use both ng-cloak and translate-cloak..

**Comment by oktay89** [comment]
@knalli Just updated to 2.8.1 - same issue:
span class="col-md-2 pageTitle" translate-cloak="gross_yield.OVERVIEW"></span

After page load and translation complete, translate-cloak class still doesnt get removed as class from my element, so it still says invisible

**Comment by tspaeth** [comment]
```
translate-cloak="gross_yield.OVERVIEW"
```

Is that your CSS class name?

**Comment by oktay89** [comment]
This is the key that is getting translated through the key-value pairs in my .json files.

**Comment by knalli** [comment]
Hm.. in combination with the partial loader and a non-existing value for the key configured in `translate-cloak`, the cloak will be removed and attached again.

Solution c/should be: In the cloak directive, add an event listener for translate changes invoked by the partial loader.

**Comment by tspaeth** [comment]
@oktay89 my fault... just didn't read carful enough... :)

**Comment by oktay89** [comment]
I use ui-router and on "$stateChangeSuccess", i load the .json file in the correct language for the selected state like this:
$translatePartialLoader.addPart(module, $translate.use());
After that:
$translate.onReady() gets called from angular-translate-cloak.js
and this function calls removeCloak(), which should remove the css class translate-cloak from all elements with translate-cloak="key" directive.
Removing class from the html elements, thats what fails in my Angular application

**Comment by knalli** [comment]
@oktay89 : I've added end of the last year following fix e7536b5. I'm not sure I'm getting your steps right.. is that matching it?

**Comment by oktay89** [comment]
Great, your fix in 2.9.0 solved the problem. Thanks!
